### PR TITLE
Feature/apply codestyle and code conventions

### PR DESCRIPTION
**Pull request by tiwalter**
Fullfill the coding conventions:
- Add source code documentation
- Adjust the spelling of the directories
- Adjust the spelling of the members
- Add regions
- Adjust access modifiers 
- Add copyright header

### CODE REVIEW

File: sources/release-server/ReleaseServer.WebApi/Program.cs [hunk 1]
```diff
@@ -1,15 +1,19 @@
...
+
 using Microsoft.AspNetCore.Hosting;
-using Microsoft.Extensions.Configuration;
 using Microsoft.Extensions.Hosting;
 using Microsoft.Extensions.Logging;
 
 namespace ReleaseServer.WebApi
 {
-    public class Program
+    internal class Program
```
Comment (f-porter) [inline sources/release-server/ReleaseServer.WebApi/Program.cs:16]: Usually `Program` is marked as `public`.

File: sources/release-server/ReleaseServer.WebApi/Startup.cs [hunk 2]
```diff
@@ -10,18 +19,23 @@
...
-using ReleaseServer.WebApi.Extensions;
-using ReleaseServer.WebApi.JsonConverters;
-using ReleaseServer.WebApi.SwaggerDocu;
+
 using Swashbuckle.AspNetCore.Filters;
 
 namespace ReleaseServer.WebApi
 {
-    public class Startup
+    internal class Startup
```
Comment (f-porter) [inline sources/release-server/ReleaseServer.WebApi/Startup.cs:29]: Usually `Startup` is marked as `public`.

**Review by f-porter — CHANGES_REQUESTED** [review]
As far as I understood the coding conventions the directory containing the solution file (`release-server`) should be named `ReleaseServer`. And I think this would be easier to associate the project name with. So please revert the change of the directory name `sources/ReleaseServer` -> `sources/release-server`.

In addition to that there is a typo in probably all of the license header of the code files. I only marked some of them as this would have been too much effort picking every single one of them. This should be fixed too.

File: Dockerfile [hunk 3]
```diff
@@ -1,3 +1,11 @@
+#--------------------------------------------------------------------------------------------------
+# <copyright file="Dockerfile" company="Traeger IndustryComponents GmbH">
```
[the remark this answers is not in the document]
Comment (tiwalter) [reply Dockerfile:2]: Done with https://github.com/Traeger-GmbH/release-server/commit/690b2b579c81f4e694ede26664f590d15870ffb3

File: sources/release-server/ReleaseServer.WebApi.Test/Common/JsonSerializeTest.cs [hunk 4]
```diff
@@ -1,25 +1,33 @@
+//--------------------------------------------------------------------------------------------------
+// <copyright file="JsonSerializeTest.cs" company="Traeger IndustryComponents GmbH">
```
[the remark this answers is not in the document]
Comment (tiwalter) [reply sources/release-server/ReleaseServer.WebApi.Test/Common/JsonSerializeTest.cs]: Done with https://github.com/Traeger-GmbH/release-server/commit/690b2b579c81f4e694ede26664f590d15870ffb3

File: sources/release-server/ReleaseServer.WebApi.Test/Extensions/DirectoryInfoExtensionTest.cs [hunk 5]
```diff
@@ -1,24 +1,29 @@
+//--------------------------------------------------------------------------------------------------
+// <copyright file="DirectoryInfoExtensionTest.cs" company="Traeger IndustryComponents GmbH">
```
[the remark this answers is not in the document]
Comment (tiwalter) [reply sources/release-server/ReleaseServer.WebApi.Test/Extensions/DirectoryInfoExtensionTest.cs]: Done with https://github.com/Traeger-GmbH/release-server/commit/690b2b579c81f4e694ede26664f590d15870ffb3

File: sources/release-server/ReleaseServer.WebApi/Extensions/ReleaseServerServiceCollectionExtension.cs [hunk 6]
```diff
@@ -1,13 +1,20 @@
+//--------------------------------------------------------------------------------------------------
+// <copyright file="ReleaseServerServiceCollectionExtension.cs" company="Traeger IndustryComponents GmbH">
+//     This file is protected by Traeger Industry Components GmbH Copyright © 2019-2020.
+// </copyright>
+// <author>Fabian Traeger</author>
+// <author>Timo Walter</author>Walter</author>
```
[the remark this answers is not in the document]
Comment (tiwalter) [reply sources/release-server/ReleaseServer.WebApi/Extensions/ReleaseServerServiceCollectionExtension.cs]: Done with https://github.com/Traeger-GmbH/release-server/commit/e1f536eacc9cae013c7419fca7c54f72705630d3

File: sources/release-server/ReleaseServer.WebApi/Handlers/BasicAuthHandler.cs [hunk 7]
```diff
@@ -1,32 +1,54 @@
...
             UrlEncoder encoder,
             ISystemClock clock) : base(options, logger, encoder, clock)
         {
-            Configuration = configuration;
+            this.configuration = configuration;
         }
+        
+        #endregion
+        
+        #region ---------- Methods ----------
```
[the remark this answers is not in the document]
Comment (tiwalter) [reply sources/release-server/ReleaseServer.WebApi/Handlers/BasicAuthHandler.cs]: Done with https://github.com/Traeger-GmbH/release-server/commit/e1f536eacc9cae013c7419fca7c54f72705630d3

File: sources/release-server/ReleaseServer.WebApi/Controllers/ReleaseArtifactController.cs [hunk 8]
```diff
@@ -0,0 +1,385 @@
...
+        #endregion
+
+        #region ---------- Public methods ----------
+        
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ReleaseArtifactController"/> class.
+        /// </summary>
+        /// <param name="logger">The <see cref="ILogger"/> to enable logging for the <see cref="ReleaseArtifactController"/>.</param>
+        /// <param name="releaseArtifactService">The service that handles the business logic of the application.</param>
+        public ReleaseArtifactController(ILogger<ReleaseArtifactController> logger,
```
[the remark this answers is not in the document]
Comment (tiwalter) [reply sources/release-server/ReleaseServer.WebApi/Controllers/ReleaseArtifactController.cs:49]: Done with https://github.com/Traeger-GmbH/release-server/commit/e1f536eacc9cae013c7419fca7c54f72705630d3

File: sources/release-server/ReleaseServer.WebApi/Repositories/FsReleaseArtifactRepository.cs [hunk 9]
```diff
@@ -1,31 +1,48 @@
...
-            BackupRootDir = backupDirectory;
-            Logger = logger;
+            artifactRootDir = artifactDirectory;
+            backupRootDir = backupDirectory;
+            this.logger = logger;
         }
+        
+        #endregion
+        
+        #region ---------- Methods ----------
```
[the remark this answers is not in the document]
Comment (tiwalter) [reply sources/release-server/ReleaseServer.WebApi/Repositories/FsReleaseArtifactRepository.cs]: Done with https://github.com/Traeger-GmbH/release-server/commit/e1f536eacc9cae013c7419fca7c54f72705630d3

File: sources/release-server/ReleaseServer.WebApi/Repositories/FsReleaseArtifactRepository.cs [hunk 10]
```diff
@@ -1,31 +1,48 @@
...
+        
+        #region ---------- Public methods ----------
```
[the remark this answers is not in the document]
Comment (tiwalter) [reply sources/release-server/ReleaseServer.WebApi/Repositories/FsReleaseArtifactRepository.cs]: Done with https://github.com/Traeger-GmbH/release-server/commit/e1f536eacc9cae013c7419fca7c54f72705630d3

File: sources/release-server/ReleaseServer.WebApi/Services/FsReleaseArtifactService.cs [hunk 11]
```diff
@@ -1,84 +1,106 @@
...
-            Logger = logger;
-            DirectoryLock = new SemaphoreSlim(1, 1);
+            this.fsReleaseArtifactRepository = fsReleaseArtifactRepository;
+            this.logger = logger;
+            directoryLock = new SemaphoreSlim(1, 1);
         }
+        
+        #endregion
+
+        #region ---------- Public methods ----------
```
[the remark this answers is not in the document]
Comment (tiwalter) [reply sources/release-server/ReleaseServer.WebApi/Services/FsReleaseArtifactService.cs]: Done with https://github.com/Traeger-GmbH/release-server/commit/e1f536eacc9cae013c7419fca7c54f72705630d3

File: Dockerfile (hunk 3 above)
[the remark this answers is not in the document]
Comment (tiwalter) [reply Dockerfile:2]: Directory name changed -> see https://github.com/Traeger-GmbH/release-server/commit/9d15b11e81cde8ee4c24b90175c0cf4e0b6b652b

File: sources/release-server/ReleaseServer.WebApi/Program.cs (hunk 1 above)
Comment (tiwalter) [reply sources/release-server/ReleaseServer.WebApi/Program.cs]: Done with https://github.com/Traeger-GmbH/release-server/pull/60/commits/2cb28c7104c2450252ad2f226c26b9bc0480c9a1

File: sources/release-server/ReleaseServer.WebApi/Startup.cs (hunk 2 above)
Comment (tiwalter) [reply sources/release-server/ReleaseServer.WebApi/Startup.cs]: Done with https://github.com/Traeger-GmbH/release-server/pull/60/commits/2cb28c7104c2450252ad2f226c26b9bc0480c9a1

**Pull request merged**